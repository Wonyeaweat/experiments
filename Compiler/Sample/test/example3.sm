program example3;
var ans, i, j: integer;
begin
    ans := 1; i := 0; j := 10;
    while i < 10000 do
       ans := ans + i;
    if i > 1000 then
       begin 
         while j > 0 do
            i := i - j
       end
end.