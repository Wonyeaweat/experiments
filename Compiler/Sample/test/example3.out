( 0) (program, example3, - , - )
( 1) (:=, 1 , - , ans)
( 2) (:=, 0 , - , i )
( 3) (:=, 10, - , j )
( 4) (j<, i , 10000, 6 )
( 5) (j , - , - , 9 )
( 6) (+ , ans, i , T1)
( 7) (:=, T1, - , ans)
( 8) (j , - , - , 4 )
( 9) (j>, i , 1000, 11)
(10) (j , - , - , 16)
(11) (j>, j , 0 , 13)
(12) (j , - , - , 16)
(13) (- , i , j , T2)
(14) (:=, T2, - , i )
(15) (j , - , - , 11)
(16) (sys, - , - , - )
